streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredKeyValueStore.java

```diff
@@ -99,9 +101,17 @@ public class MeteredKeyValueStore<K, V>
                 RangeQuery.class,
                 (query, positionBound, config, store) -> runRangeQuery(query, positionBound, config)
             ),
+            mkEntry(
+                TimestampedRangeQuery.class,
```

Comment:
Why does `MeteredKeyValueStore` implement this? I thought this would only go into `MeteredTimestampKeyValueStore`?

Reply:
Currently, implementing this method in `MeteredTimestampKeyValueStore `requires copying helper functions from `MeteredKeyValueStore`. However, by implementing it in `MeteredKeyValueStore`, we can directly utilize these helper functions and achieve code reuse.

Reply:
Not sure what helper function you mean (we call the newly added `runTimestampRangeQuery`)?

But even if there would be a code-reuse problem, it's just the wrong place. `MeteredKeyValueStore` is for plain-kv-store, and should only implement what plain-kv-store supports.

Reply:
ok